src/RobotFrames.hpp

```diff
@@ -210,6 +211,48 @@ class TransformationCalculator
     bool output_only_valid_;
 };
 
+class ChainTransformationCalculator{
+
+protected:
+    std::vector<robot_frames::Chain> Chains_;
+    std::string urdf_file_; 
+    base::samples::Joints joint_state_;
+
+    // variables from the orogen-component to mimic the functionality 
+    std::vector<KDL::Chain> chains_;
+    std::vector<std::string> chain_names_;
+    std::vector<KDL::ChainFkSolverPos_recursive*> pos_solvers_;
+    std::vector<KDL::Frame> kdl_frames_;
+    std::vector<base::samples::RigidBodyState> bt_frames_;
+    std::vector<KDL::JntArray> joint_arrays_;
+    std::vector<std::vector<std::string> >involved_active_joints_;
+    KDL::Tree tree_;
+    int n_defined_chains_;
+
+public:
+
+    void initilize(std::vector<robot_frames::Chain> chain, std::string urdf, base::samples::Joints jointstate);
+    
+    void clear_and_resize_vectors();
+    void unpack_joints(const base::samples::Joints& joint_state,
```

Comment:
No need for being exposed as public, or? I'd prefer to make it protected then. I guess with the proposed signature this function does not modify and member variables?!?
If so, consider moving it out of the class or make it static.

Reply:
unpack_joints() function wont modify member variables but clear_and_resize_vectors() will. 
yes, it can be protected.